sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java

```diff
@@ -187,18 +194,23 @@ public T getValue() {
     return windowedValues.build();
   }
 
-  /**
-   * Returns the pane of this {@code WindowedValue} in its window.
-   */
-  public PaneInfo getPane() {
-    return pane;
-  }
-
   @Override
-  public abstract boolean equals(Object o);
+  public boolean equals(Object other) {
+    if (!(other instanceof WindowedValue)) {
+      return false;
+    } else {
+      WindowedValue<?> that = (WindowedValue<?>) other;
+      return Objects.equals(this.getValue(), that.getValue())
```

Comment:
I think we often compared timestamps first (more likely to be different), and you likely want to do it in terms of the millis (otherwise it goes through some chronology, which makes them slow). This is important since these may be used in the maps for doing precombine, etc.